predicators/envs/spot_env.py

```diff
@@ -765,10 +887,28 @@ def render_state_plt(
         raise NotImplementedError("This env does not use Matplotlib")
 
     def _load_task_from_json(self, json_file: Path) -> EnvironmentTask:
-        # Use the BaseEnv default code for loading from JSON, which will
-        # create a State as an observation. We'll then convert that State
-        # into a _SpotObservation instead.
-        base_env_task = super()._load_task_from_json(json_file)
+        with open(json_file, "r", encoding="utf-8") as f:
+            json_dict = json.load(f)
+        object_name_to_object = self._parse_object_name_to_object_from_json(
+            json_dict)
+        init_dict = self._parse_init_state_dict_from_json(
+            json_dict, object_name_to_object)
+        # Get the object detection id's, which are features of
+        # each object type.
+        obj_to_detection_id = {
+            v: k
+            for k, v in self._detection_id_to_obj.items()
+        }
+        for obj, init_val in init_dict.items():
+            if obj in obj_to_detection_id:
+                init_val["object_id"] = obj_to_detection_id[
```

Comment:
what exactly is the type issue here?

Reply:
the issue is that the `init_dict` expects to be a mapping from objects to a dict of features and floats, but for spot envs, one of the features of an object is an `ObjectDetectionID`, which is not a float